nixos/modules/services/misc/paperless.nix

```diff
@@ -248,11 +247,34 @@ in
         ```
       '';
     };
+
+    database = {
+      createLocally = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Configure local PostgreSQL database server for Paperless.
+        '';
+      };
+    };
   };
 
   config = mkIf cfg.enable {
     services.redis.servers.paperless.enable = mkIf enableRedis true;
 
+    services.postgresql = lib.mkIf cfg.database.createLocally {
+      enable = true;
+      ensureDatabases = [ "paperless" ];
+      ensureUsers = [{
+        name = config.services.paperless.user;
+        ensureDBOwnership = true;
+      }];
+    };
+
+    services.paperless.settings = lib.mkIf cfg.database.createLocally {
+      PAPERLESS_DBHOST = "/run/postgresql";
```

Comment:
Perhaps should also include `PAPERLESS_DBNAME = "paperless"`?

Reply:
Added all of `PAPERLESS_DBENGINE`, `PAPERLESS_DBNAME`, `PAPERLESS_DBUSER` as well.